src/Proxy/CommonServiceProxy.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Yiisoft\Yii\Debug\Proxy;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\Proxy\ObjectProxy;
+use Yiisoft\Yii\Debug\Collector\CommonServiceCollectorInterface;
+use Yiisoft\Yii\Debug\Event\ProxyMethodCallEvent;
+
+class CommonServiceProxy extends ObjectProxy
+{
+    const LOG_ARGUMENTS = 1 << 0;
+
+    const LOG_RESULT = 1 << 1;
+
+    const LOG_ERROR = 1 << 2;
+
+    private string $service;
+
+    private $logLevel = 0;
+
+    private ?CommonServiceCollectorInterface $collector = null;
+
+    private ?EventDispatcherInterface $dispatcher = null;
+
+    public function __construct(
+        string $service,
+        object $instance,
+        CommonServiceCollectorInterface $collector = null,
+        EventDispatcherInterface $dispatcher = null,
+        int $logLevel = 0
+    ) {
+        $this->service = $service;
+        $this->collector = $collector;
+        $this->dispatcher = $dispatcher;
+        $this->logLevel = $logLevel;
+        parent::__construct($instance);
+    }
+
+    protected function executeMethodProxy(string $methodName, array $arguments, $result, float $timeStart)
+    {
+        $this->log($methodName, $arguments, $result, $timeStart);
+        return $result;
+    }
+
+    protected function getNewStaticInstance(object $instance): ObjectProxy
+    {
+        return new static($this->service, $instance, $this->collector, $this->dispatcher, $this->logLevel);
+    }
+
+    protected function log(string $method, array $arguments, $result, float $timeStart): void
+    {
+        $error = $this->getCurrentError();
+        $this->processLogData($arguments, $result, $error);
+
+        if ($this->collector !== null) {
+            $this->logToCollector($method, $arguments, $result, $error, $timeStart);
+        }
+
+        if ($this->dispatcher !== null) {
+            $this->logToEvent($method, $arguments, $result, $error, $timeStart);
+        }
+    }
+
+    private function processLogData(array &$arguments, &$result, ?object &$error): void
+    {
+        if (!($this->logLevel & self::LOG_ARGUMENTS)) {
+            $arguments = null;
+        }
+
+        if (!($this->logLevel & self::LOG_RESULT)) {
+            $result = null;
+        }
+
+        if (!($this->logLevel & self::LOG_ERROR)) {
+            $error = null;
+        }
+    }
+
+    protected function getService(): string
+    {
+        return $this->service;
+    }
+
+    protected function getCollector(): CommonServiceCollectorInterface
+    {
+        return $this->collector;
+    }
+
+    protected function getDispatcher(): EventDispatcherInterface
+    {
+        return $this->dispatcher;
+    }
+
+    protected function getLogLevel(): int
+    {
+        return $this->logLevel;
+    }
+
+    private function logToCollector(string $method, ?array $arguments, $result, ?object $error, float $timeStart): void
```

Comment:
`collector` can't receive $arguments as a `null`

Reply:
It can

Reply:
https://github.com/yiisoft/yii-debug/pull/35/files#diff-101632852038c6777642bec52f664632R11